micrometer-tracing-bridges/micrometer-tracing-bridge-brave/src/main/java/io/micrometer/tracing/brave/bridge/BraveTracer.java

```diff
@@ -44,6 +44,9 @@ public BraveTracer(brave.Tracer tracer, CurrentTraceContext context, BaggageMana
         this.tracer = tracer;
         this.braveBaggageManager = braveBaggageManager;
         this.currentTraceContext = context;
+        if (braveBaggageManager instanceof BraveBaggageManager) {
```

Comment:
If it's not a BraveBaggageManager, isn't it an error? Should the previous assignment still be executed?

Reply:
Theoretically a user can provide their own baggage manager. Practically there are almost zero chances that this would happen